src/pages/ListSwaps/LedgerActionDialogBody.tsx

```diff
@@ -108,15 +115,19 @@ function LedgerActionDialogBody({
         </React.Fragment>
       );
     }
-    case "ethereum-invoke-contract": {
+    case "ethereum-call-contract": {
+      const expiry = action.payload.min_block_timestamp;
+      const whenReadyMessage =
```

Comment:
For Ethereum, you would have to compare it to the blockchain's timestamp as-well.

For that, you would have to ask `web3` for the latest block and check the `timestamp` property.

To align this requirement with my comment above about disabling the `Web3SendTransactionButton`, I would add the functionality to the `Web3SendTransactionButton` itself. Then you can satisfy both requirements:
- Not having a 2nd source for when the button is disabled
- Disabling the button correctly based on the time

Since you can implement this inside the `Web3SendTransactionButton`, you just need to pass the `min_block_timestamp` value to it and do some Web3 calls in there to figure out whether it is already passed that time. You can then also modify the tooltip text accordingly (initially disabled until web3 is available and, if you got passed a `timestamp`, the tooltip should say that the current blocktime is and when the action will be available.)

Let me know if you need more help on this :)

Reply:
This is a good idea, I'll give it a go!